javascript/javascript2/week3/warmup.js

```diff
@@ -0,0 +1,153 @@
+//1.Log out the text Called after 2.5 seconds 2.5 seconds after the script is loaded.
+console.log("Script loaded")
+setTimeout(() => console.log("Called after 2.5 seconds"), 2500);
+
+//2. Create a function that takes 2 parameters: delay and stringToLog. 
+//Calling this function should log out the stringToLog after delay seconds. 
+//Call the function you have created with some different arguments.
+
+function logDelayedString(delaySeconds, stringToLog) {
+    setTimeout(() => console.log(stringToLog), delaySeconds * 1000);
+}
+
+logDelayedString(1, "Hello after 1 sec");
+logDelayedString(2, "World after 2 sec");
+
+//3.Create a button in html. When clicking this button, use the function you created in the 
+//previous task to log out the text: `Called after 5 seconds` 5 seconds after the button is clicked.
+
+const btn = document.querySelector("#button3");
+btn.addEventListener("click", delayedClick(5, "Called after 5 seconds"));
+
+//creating calback with passed arguments
+function delayedClick(delaySeconds, stringToLog) {
+    function delayCallback() {
+        logDelayedString(delaySeconds, stringToLog);
+    }
+    return delayCallback;
+}
+
+//4.Create two functions and assign them to two different variables. 
+//One function logs out Earth, the other function logs out Saturn. 
+//Now create a new third function that has one parameter: planetLogFunction. 
+//The only thing the third function should do is call the provided parameter function. 
+//Try call the third function once with the Earth function and once with the Saturn function.
+
+const earthLogger = () => console.log("Earth");
+const saturnLogger = () => console.log("Saturn");
+
+function planetLogFunction(callback) {
+    return callback();
+}
+
+planetLogFunction(earthLogger);
+planetLogFunction(saturnLogger);
+
+//Create a button with the text called "Log location". 
+//When this button is clicked the location (latitude, longitude) 
+//of the user should be logged out using this browser api
+
+const btn2 = document.querySelector("#location");
+btn2.addEventListener("click", logLocation);
+function logLocation() {
+    function success(pos) {
+        const latitude = document.querySelector("#lat");
+        latitude.innerHTML = `This is the Latitude : ${pos.coords.latitude}`;
+        const longitude = document.querySelector("#long");
+        longitude.innerHTML = `This is the Longitude : ${pos.coords.longitude}`;
+
+        //6.Optional Now show that location on a map using fx the Google maps api
+        // I attach the script for google.maps callback dynamically to work only after button Location pressed
+        // Create the script tag, set the appropriate attributes
+        var script = document.createElement('script');
+        script.src = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyDFGlu0GRChxbWdneIqV7lkJrS8B3Hw-ts&callback=initMap';
```

Comment:
This is not a problem in this case, but be careful about having tokens directly in the code